runtime/lf_skiplist.c

```diff
@@ -353,8 +353,10 @@ int caml_lf_skiplist_insert(struct lf_skiplist *sk, uintnat key, uintnat data) {
     struct lf_skipcell *succ;
 
     if (found) {
-      /* node already exists, we don't need to do anything */
-      return 0;
+      /* Already present; update data */
+      atomic_store_explicit((atomic_uintnat*)&succs[0]->data, data,
+                            memory_order_release);
```

Comment:
I don't think it's safe to cast a `uintnat` to an `atomic_uintnat` so we will need to make data an `atomic_uintnat`.

Also the stores can be relaxed, we never do a release on `data` (we do the acquire/release on the forward references at each level in insert/removal).

Reply:
Done!

Reply:
Thanks, this looks fine.

For the cast, it looks like we're doing this in a few other places and assuming we can cast between words and atomic words anyway (e.g https://github.com/ocaml/ocaml/blob/905d17c95ddadfc919d34ef156f656bde8939b55/runtime/minor_gc.c#L637).